test/project_types/extension/models/specification_handlers/checkout_ui_extension_test.rb

```diff
@@ -218,6 +218,34 @@ def test_l10n_sum_max_filesize
           end
         end
 
+        def test_l10n_invalid_file_encoding
+          write("locales/en.default.json", '{"laugh": "lol"}', encoding: Encoding::UTF_16)
+          assert_raises ShopifyCLI::Abort, "#{L10N_ERROR_PREFIX}.invalid_file_encoding" do
+            @checkout_ui_extension.config(@context)
+          end
+        end
+
+        def test_l10n_utf8_bom_is_stripped
+          en_content = '{"laugh": "lol"}'
+
+          filename = File.join(@context.root, "locales/en.default.json")
+          FileUtils.mkdir_p(File.dirname(filename))
+          File.open(filename, "w", 0644) do |file|
+            file.write("\xEF\xBB\xBF")
```

Comment:
I had an issue before that led me to this but I think I figured it out and can simplify this test back to look like the others (using the `write` helper)